### PR TITLE
Adds 5 new scroll items with unique effects

### LINKED ISSUE #6091: voice volume not distance dependent
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context

- Minecraft Version: 1.16.4
- MineColonies Version: .13.444

## Expected behavior
voices play at a volume dependent on the distance from the player.

## Actual behavior
the voice files play at the same volume now matter how far away. so long as I am close enough to trigger the colonists talking, they talk at full volume instead of it being quieter because they are a chunk away.

## Steps to reproduce the issue

get close enough to trigger a colonist talking, but far enough away it should be quiet. and then compare with the volume at 1 block instead of 16+

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by someaddons**
Closes #6091

# Changes proposed in this pull request:
Adds 5 new scroll items with unique effects:
A colony home teleportation scroll, which teleports the user to the registered colony. 
A colony home area teleportation scroll which teleports the user and nearby players to the colony.
A regeneration scroll, granting regeneration to the player and citizens around him, found in chests.
A guard help scroll which temporarily summons the guards of a registered guard building to the user on follow mode.
A worker highlight scroll which can be used on a worker building an highlights the worker with a glow effect and gives him a slight speed effect too.
image

Fix some of our sounds beeing stereo and not playing positioned
Review please

### CODE REVIEW

**Review by Raycoms — DISMISSED** [review]

**Comment by ravenbuilder934** [comment]
> Fix some of our sounds beeing stereo and not playing positioned

Does this close #6091?

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 1]
```diff
@@ -338,6 +338,33 @@
   "item.minecolonies.golden_bread": "Golden Bread",
   "item.minecolonies.chorus_bread": "Chorus Fruit Bread",
 
+  "item.minecolonies.scroll_tp": "Ultrasafe colony teleport scroll",
+  "item.minecolonies.scroll_tp.tip": "Safely teleports you to the set colony",
+  "item.minecolonies.scroll.colony.tip": "Colony: %s",
+  "item.minecolonies.scroll.colony.none": "None, set one with sneak rightclick",
+  "item.minecolonies.scroll_area_tp": "Ultrasafe colony group-teleport scroll",
+  "item.minecolonies.scroll_area_tp.tip": "Teleport yourself and nearby players to the colony",
+  "item.minecolonies.scroll_buff": "Sacred scroll of Regeneration",
```
Comment (ravenbuilder934) [inline src/main/resources/assets/minecolonies/lang/en_us.json]:
```suggestion
  "item.minecolonies.scroll_buff": "Sacred Scroll of Regeneration",
```

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 2]
```diff
@@ -338,6 +338,33 @@
   "item.minecolonies.golden_bread": "Golden Bread",
   "item.minecolonies.chorus_bread": "Chorus Fruit Bread",
 
+  "item.minecolonies.scroll_tp": "Ultrasafe colony teleport scroll",
+  "item.minecolonies.scroll_tp.tip": "Safely teleports you to the set colony",
+  "item.minecolonies.scroll.colony.tip": "Colony: %s",
+  "item.minecolonies.scroll.colony.none": "None, set one with sneak rightclick",
```
Comment (ravenbuilder934) [inline src/main/resources/assets/minecolonies/lang/en_us.json]:
```suggestion
  "item.minecolonies.scroll.colony.none": "None, set one with sneak + right-click",
```

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 3]
```diff
@@ -338,6 +338,33 @@
...
+  "item.minecolonies.scroll_area_tp": "Ultrasafe colony group-teleport scroll",
+  "item.minecolonies.scroll_area_tp.tip": "Teleport yourself and nearby players to the colony",
+  "item.minecolonies.scroll_buff": "Sacred scroll of Regeneration",
+  "item.minecolonies.scroll_buff.tip": "Applies regeneration to citizens around you, at least most of the times...",
```
Comment (ravenbuilder934) [inline src/main/resources/assets/minecolonies/lang/en_us.json]:
```suggestion
  "item.minecolonies.scroll_buff.tip": "Applies regeneration to citizens around you, at least most of the time...",
```

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 4]
```diff
@@ -338,6 +338,33 @@
...
+  "item.minecolonies.scroll_guard_help": "Spatial guard reinforcement scroll",
+  "item.minecolonies.scroll_guard_help.tip": "Call your guards to help you out for a limited time, wololo!",
+  "item.minecolonies.scroll_highlight": "Worker-where-are-you scroll",
+  "item.minecolonies.scroll_highlight.tip": "Sneak-Use on a minecolonies building, to highlight and speed up its workers",
+  "minecolonies.scroll.nopermission": "You do not have the required permissions for this colony.",
+  "minecolonies.scroll.needcolony": "Please register this to a colony first with shift-rightclicking a building.",
```
Comment (ravenbuilder934) [inline src/main/resources/assets/minecolonies/lang/en_us.json]:
```suggestion
  "minecolonies.scroll.needcolony": "Please register this to a colony first with shift + right-clicking a building.",
```

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 5]
```diff
@@ -338,6 +338,33 @@
...
+  "minecolonies.scroll.noguardbuilding": "Please register this to a guard building by shift-rightclicking it.",
```
Comment (ravenbuilder934) [inline src/main/resources/assets/minecolonies/lang/en_us.json]:
```suggestion
  "minecolonies.scroll.noguardbuilding": "Please register this to a guard building by shift + right-clicking it.",
```

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 6]
```diff
@@ -338,6 +338,33 @@
...
+  "minecolonies.scroll.wrongdim": "Can only be used in the same dimension as the colony.",
+  "minecolonies.scroll.registered": "Registered scroll with colony %s",
+  "minecolonies.scroll.failed1": "The unstable magic of the scroll backfired!",
+  "minecolonies.scroll.failed2": "Whoops, what happened now?!",
```
Comment (ravenbuilder934) [inline src/main/resources/assets/minecolonies/lang/en_us.json:359]:
```suggestion
  "minecolonies.scroll.failed2": "Oopsies!",
```

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 7]
```diff
@@ -338,6 +338,33 @@
...
+  "minecolonies.scroll.failed3": "This scroll needs to go back to the drawing board!",
```
Comment (ravenbuilder934) [inline src/main/resources/assets/minecolonies/lang/en_us.json:360]:
```suggestion
  "minecolonies.scroll.failed3": "Uh oh!",
```

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 8]
```diff
@@ -338,6 +338,33 @@
...
+  "minecolonies.scroll.failed4": "The scroll doesnt seem to be made by the best magician out there!",
```
Comment (ravenbuilder934) [inline src/main/resources/assets/minecolonies/lang/en_us.json:361]:
```suggestion
  "minecolonies.scroll.failed4": "Sorry about that!",
```

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 9]
```diff
@@ -338,6 +338,33 @@
...
+  "minecolonies.scroll.failed5": "Well, this is embarrassing.",
+  "minecolonies.scroll.failed6": "Looks like there's a typo in one of the ancient glyphs!",
+  "minecolonies.scroll.failed7": "Did my I get the wrong spell again?!",
```
Comment (ravenbuilder934) [inline src/main/resources/assets/minecolonies/lang/en_us.json]:
```suggestion
  "minecolonies.scroll.failed7": "Did my creator copy the wrong spell again?!",
```

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 10]
```diff
@@ -338,6 +338,33 @@
...
+  "minecolonies.scroll.failed8": "Wasn't this supposed to do something?",
```
Comment (ravenbuilder934) [inline src/main/resources/assets/minecolonies/lang/en_us.json:365]:
```suggestion
  "minecolonies.scroll.failed8": "Oh, was I supposed to do something?",
```

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 11]
```diff
@@ -338,6 +338,33 @@
...
+  "minecolonies.scroll.failed9": "404 Spell Not Found",
+  "minecolonies.scroll.failed10": "Maybe try some ladders instead?",
```
Comment (ravenbuilder934) [inline src/main/resources/assets/minecolonies/lang/en_us.json:367]:
```suggestion
  "minecolonies.scroll.failed10": "Maybe try some ladders instead?",
  "minecolonies.scroll.failed11": "I blame Someaddons.",
  "minecolonies.scroll.failed12": "I blame Wissi.",
  "minecolonies.scroll.failed13": "418 I'm a teapot, not a spell!",
  "minecolonies.scroll.failed14": "You should try our sister game, ColonyColonyColonyColonyColonyColonyColonyColonyColony!",
```

File: src/main/java/com/minecolonies/coremod/items/AbstractItemScroll.java [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+import static com.minecolonies.api.util.constant.NbtTagConstants.TAG_COLONY_ID;
+
+/**
+ * Scroll items base class, does colony registering/checks.
+ */
+public abstract class AbstractItemScroll extends AbstractItemMinecolonies
+{
+    public static final String TAG_COLONY_DIM       = "colony_dim";
+    public static final String TAG_BUILDING_POS     = "building_pos";
+    public static final int    FAIL_RESPONSES_TOTAL = 10;
```
Comment (ravenbuilder934) [inline src/main/java/com/minecolonies/coremod/items/AbstractItemScroll.java:34]:
```suggestion
    public static final int    FAIL_RESPONSES_TOTAL = 14;
```

**Review by ravenbuilder934 — CHANGES_REQUESTED** [review]

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingEnchanter.java [hunk 13]
```diff
@@ -33,7 +34,7 @@
 /**
  * The enchanter building.
  */
-public class BuildingEnchanter extends AbstractBuildingWorker
+public class BuildingEnchanter extends AbstractBuildingCrafter
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingEnchanter.java:38]: This building was turned into a crafter. Yet no methods where overriden. Sooo is this a default crafter? I think you would like ant to limit the recipes would you not?

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingEnchanter.java:38]: @OrionDevelopment afaik that is all data driven now. @Mekle001 this should be fine as is, right?

Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingEnchanter.java:38]: hm good point I'll go ahead and test it

Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingEnchanter.java:38]: was able to teach it sticks xD

**Pull request merged**